pkg/container/resolver_test.go

```diff
@@ -114,3 +117,82 @@ func TestResolverFail(t *testing.T) {
 	assert.Error(t, err)
 	assert.Len(t, specs, 0)
 }
+
+func TestResolverLocalManifest(t *testing.T) {
+	currentUser, err := user.Current()
+	if err != nil {
+		t.Fatalf("Failed to get current user: %v", err)
+	}
+
+	if currentUser.Uid != "0" {
+		t.Skip("User is not root, skipping test")
+	}
+
+	_, err = exec.LookPath("podman")
+	if err != nil {
+		t.Skip("Podman not available, skipping test")
+	}
+
+	containerFile, err := os.CreateTemp("", "Containerfile")
+	if err != nil {
+		t.Fatalf("Failed to create tempfile: %v", err)
+	}
+	defer os.Remove(containerFile.Name())
+
+	// we need something more than scratch otherwise
+	// two containers with different arches will share
+	// the same underlying container image
+	_, err = containerFile.Write([]byte("FROM fedora-minimal\nRUN uname -m"))
```

Comment:
To do the whole `podman build --platform ...` thing I had to use a `fedora-minimal` base because if we just used `scratch` it would only build a single container that could be used for both arches.

Reply:
I was thinking we could use scratch and add an argument that changes the two based on the architecture, like I described in https://github.com/osbuild/images/pull/595#pullrequestreview-2015155890, but we can do that in a followup.